test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -212,34 +218,242 @@ func (s *snapshottableTestSuite) DefineTests(driver TestDriver, pattern testpatt
+			framework.ExpectNoError(dc.Resource(SnapshotClassGVR).Delete(context.TODO(), vsc.GetName(), metav1.DeleteOptions{}))
+		}()
+		snapshotClass, err := dc.Resource(SnapshotClassGVR).Namespace(vsc.GetNamespace()).Get(context.TODO(), vsc.GetName(), metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		snapshotClass.Object["deletionPolicy"] = "Retain"
+
+		ginkgo.By("creating a snapshot")
+		snapshot := getSnapshot(pvc.Name, pvc.Namespace, vsc.GetName())
+
+		snapshot, err = dc.Resource(SnapshotGVR).Namespace(snapshot.GetNamespace()).Create(context.TODO(), snapshot, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		defer func() {
+			framework.Logf("deleting snapshot %q/%q", snapshot.GetNamespace(), snapshot.GetName())
+			// typically this snapshot has already been deleted
+			err = dc.Resource(SnapshotGVR).Namespace(snapshot.GetNamespace()).Delete(context.TODO(), snapshot.GetName(), metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				framework.Failf("Error deleting snapshot %q. Error: %v", pvc.Name, err)
+			}
+		}()
+		err = WaitForSnapshotReady(dc, snapshot.GetNamespace(), snapshot.GetName(), framework.Poll, framework.SnapshotCreateTimeout)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("checking the snapshot")
+		// Get new copy of the snapshot
+		snapshot, err = dc.Resource(SnapshotGVR).Namespace(snapshot.GetNamespace()).Get(context.TODO(), snapshot.GetName(), metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		// Get the bound snapshotContent
+		snapshotStatus := snapshot.Object["status"].(map[string]interface{})
+		snapshotContentName := snapshotStatus["boundVolumeSnapshotContentName"].(string)
+		snapshotContent, err := dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("checking the SnapshotContent")
+		snapshotContentSpec := snapshotContent.Object["spec"].(map[string]interface{})
+		volumeSnapshotRef := snapshotContentSpec["volumeSnapshotRef"].(map[string]interface{})
+		framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], vsc.GetName())
+		framework.ExpectEqual(snapshotContentSpec["deletionPolicy"], "Retain")
+		framework.ExpectEqual(volumeSnapshotRef["name"], snapshot.GetName())
+		framework.ExpectEqual(volumeSnapshotRef["namespace"], snapshot.GetNamespace())
+
+		ginkgo.By("deleting the snapshot")
+		err = dc.Resource(SnapshotGVR).Namespace(snapshot.GetNamespace()).Delete(context.TODO(), snapshot.GetName(), metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			framework.Failf("Error deleting snapshot %q. Error: %v", pvc.Name, err)
+		}
+
+		ginkgo.By("checking the SnapshotContent has not been deleted")
+		_, err = dc.Resource(SnapshotContentGVR).Get(context.TODO(), snapshotContentName, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+
+		if strings.Contains(driver.GetDriverInfo().Name, "hostpath") {
```

Comment:
Can you add some comments for this section of the code to explain what it is trying to achieve?

Reply:
Added